Geometry/VeryForwardGeometryBuilder/interface/MeasuredGeometryProducer.h

```diff
@@ -0,0 +1,260 @@
+/****************************************************************************
+*
+* Authors:
+*  Jan Kaspar (jan.kaspar@gmail.com) 
+*  Dominik Mierzejewski <dmierzej@cern.ch>
+*    
+****************************************************************************/
+
+#include "FWCore/Framework/interface/ESHandle.h"
+ 
+#include "DetectorDescription/Core/interface/DDCompactView.h"
+#include "DetectorDescription/Core/interface/graphwalker.h"
+#include "DetectorDescription/Base/interface/DDRotationMatrix.h"
+#include "DetectorDescription/Core/src/LogicalPart.h"
+
+#include "DataFormats/CTPPSAlignment/interface/RPAlignmentCorrectionsData.h"
+#include "DataFormats/CTPPSDetId/interface/TotemRPDetId.h"
+
+#include <TMatrixD.h>
+
+// Copies ideal_ddcv to measured_ddcv, applying the alignments if any
+// WARNING: (TODO?) does not handle the "old" geometry
+class MeasuredGeometryProducer
+{
+  private:
+    const DDCompactView &idealCV;
+    const RPAlignmentCorrectionsData *const alignments;
+    DDLogicalPart root;
+    DDCompactView *measuredCV;
+
+    // -- Expanded view utilits ------------
+    // WTF WARNING
+    // DDExpandView's constructor sets DDRotation::StoreT to readonly
+    //
+    // Any newExpandedView() call will set this value before calling the constructor
+    // Any delExpandedView() call will the restore StoreT state to _the_last_ set value
+    static bool evRotationStoreState;
```

Comment:
Yes, one could replace this line:
https://github.com/CTPPS/cmssw/blob/CMSSW_8_1_X/Geometry/VeryForwardGeometryBuilder/plugins/TotemRPGeometryESModule.cc#L451
with a copy of the `DDCompactView` structure. In turn, this would hide the only use of `MeasuredGeometryProduce` here:
https://github.com/CTPPS/cmssw/blob/CMSSW_8_1_X/Geometry/VeryForwardGeometryBuilder/plugins/TotemRPGeometryESModule.cc#L416